test/integration/client/exec_test.go

```diff
@@ -657,22 +647,50 @@ func createNamespace(ctx context.Context, t *testing.T, client clientset.Interfa
 	return ns
 }
 
-func startConfigMapInformer(ctx context.Context, t *testing.T, client clientset.Interface, namespace string) *informerSpy {
+func startConfigMapInformer(ctx context.Context, t *testing.T, client clientset.Interface, namespace string) (cache.SharedIndexInformer, *informerSpy) {
 	t.Helper()
 
 	var informerSpy informerSpy
 	informerFactory := informers.NewSharedInformerFactoryWithOptions(client, 0, informers.WithNamespace(namespace))
-	informerFactory.Core().V1().ConfigMaps().Informer().AddEventHandler(&informerSpy)
-	informerFactory.Start(ctx.Done())
-	synced := informerFactory.WaitForCacheSync(ctx.Done())
-	if len(synced) != 1 {
-		t.Fatalf("expected only 1 synced type, got %v", synced)
+	cmInformer := informerFactory.Core().V1().ConfigMaps().Informer()
+	cmInformer.AddEventHandler(&informerSpy)
+	if err := cmInformer.SetWatchErrorHandler(func(r *cache.Reflector, err error) {
+		// t.Logf("watch error handler: failure in reflector %#v: %v", r, err) // Uncomment for more verbose logging
+	}); err != nil {
+		t.Fatalf("could not set watch error handler: %v", err)
 	}
-	if cmSynced, ok := synced[reflect.TypeOf(&corev1.ConfigMap{})]; !(cmSynced && ok) {
-		t.Fatalf("expected ConfigMaps to be synced, got %v", synced)
+	informerFactory.Start(ctx.Done())
+
+	return cmInformer, &informerSpy
+}
+
+func waitForInformerSync(t *testing.T, informer cache.SharedIndexInformer, wantSynced bool, lastSyncResourceVersion string) {
+	t.Helper()
+
+	err := wait.PollImmediate(time.Second, time.Second*30, func() (bool, error) {
+		return informer.HasSynced(), nil
+	})
+	if wantSynced {
+		if err != nil {
+			t.Fatalf("wanted sync, but got error: %v", err)
+		}
+	} else {
+		if !errors.Is(err, wait.ErrWaitTimeout) {
+			if err != nil {
+				t.Fatalf("wanted no sync, but got error: %v", err)
+			} else {
+				t.Fatalf("wanted no sync, but got sync")
+			}
+		}
```

Comment:
I was originally using `WaitForCacheSync`, but switched to this way because...I can't remember? I'll switch back to `WaitForCacheSync` until I remember why I did it this more complicated way...I should really comment my code more..._shakes fist at my time in a previous organization where I picked up this habit_. Fixed here: 3b447d4a53a.